plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/TrinoRow.java

```diff
@@ -59,4 +82,84 @@ public String toString()
     {
         return "TrinoRow" + Arrays.toString(values);
     }
+
+    static Object getObjectValue(ConnectorSession session, Block block, int position, Type type)
```

Comment:
Can we move this logic to IcebergPageSink, maybe an overload with `getIcebergValue`, or even maybe we could extract a util